boa/src/lib.rs

```diff
@@ -71,15 +74,23 @@ pub fn forward(engine: &mut Interpreter, src: &str) -> String {
 /// The str is consumed and the state of the Interpreter is changed
 /// Similar to `forward`, except the current value is returned instad of the string
 /// If the interpreter fails parsing an error value is returned instead (error object)
+#[allow(clippy::unit_arg, clippy::drop_copy)]
```

Comment:
Just for my understanding, why is this needed here? Is the timer `Copy` or the profiler?

Reply:
```
note: argument has type ()
  --> boa\src\lib.rs:90:10
   |
90 |     drop(main_timer);
```

So it looks like `()` implements Copy and when we don't compile the profiler, we're just dropping `()` which triggers this linting warning.

Reply:
This could be solved by only calling `drop()` with the feature active. We can add an artificial scope here and add the `#[cfg]` flag.